src/neptune_lightgbm/impl/__init__.py

```diff
@@ -372,15 +375,17 @@ def create_booster_summary(
             ax.imshow(image.imread(s))
             ax.axis("off")
             trees_series.append(neptune.types.File.as_image(ax.figure))
-        results_dict["{}trees".format(visuals_path)] = neptune.types.FileSeries(trees_series)
+        results_dict[f"{visuals_path}trees"] = neptune.types.FileSeries(trees_series)
 
     if log_trees_as_dataframe:
         if isinstance(booster, lgb.Booster):
             df = booster.trees_to_dataframe()
-            html_df = neptune.types.File.as_html(df)
-            results_dict["trees_as_dataframe"] = html_df
-            if not df.empty and not html_df.content:
-                warnings.warn("'trees_as_dataframe' wasn't logged. Probably generated dataframe was to large.")
+            stream_buffer = BytesIO()
+            df.to_csv(stream_buffer, index=False)
+            try:
+                results_dict["trees_as_dataframe"] = neptune.types.File.from_stream(stream_buffer, extension="csv")
+            except ValueError:
+                warnings.warn("'trees_as_dataframe' is larger than 32MB and won't be logged.")
```

Comment:
@Raalsky - does [@neptune-ai/neptune-client#1070 ](https://github.com/neptune-ai/neptune-client/releases/tag/0.16.12) make this obsolete?

Reply:
Probably yes but on the other hand, there was a bug that may impact this behavior. Let's wait till `0.16.13` :wink: